### DISCUSSION TITLE
freebsd unsupported platform

### DISCUSSION THREAD

**Discussion by theNoahWilkes** [jef/streetmerchant#1131] [Help] [answered]
I'm attempting to run streetmerchant in a freebsd jail.  I'm able to install with 'npm install' (after installing prerequisites) but 'npm run start' fails with the following error message:

`root@snatcher:~/streetmerchant # npm run start

> streetmerchant@ start /root/streetmerchant
> npm run build && node build/index.js

> build
> tsc

ＳＴＲＥＥＴＭＥＲＣＨＡＮＴ
3.1.0

[11:04:29 PM] warn :: nvidia is deprecated in favor of bestbuy
[11:04:29 PM] info :: ℹ selected stores: nvidia
[11:04:29 PM] info :: ℹ selected series: 3060ti, 3070, 3080, 3090, rx6800, rx6800xt, rx6900xt, ryzen5600, ryzen5800, ryzen5900, ryzen5950, sonyps5c, sonyps5de, xboxss, xboxsx
[11:04:32 PM] error :: ✖ something bad happened, resetting streetmerchant in 5 seconds Unsupported platform: freebsd {`

Is this a fool's errand?

**Comment by ccurtis** [comment, marked as answer]
@sixN I got this to work within my jail by installing chromium and then setting the execution path to the local binary. 

```bash
pkg install chromium
```

Add `executablePath: '/usr/local/bin/chrome'` to the puppeteer launch within src/index.ts. 

```javascript
browser = await puppeteer.launch({
    args,
    defaultViewport: {
        height: config.page.height,
        width: config.page.width
    },
    headless: config.browser.isHeadless,
    executablePath: '/usr/local/bin/chrome', //add this line
});
```

**Reply by jef** [reply]
You can also do this by using `PUPPETEER_EXECUTABLE_PATH`. I should put this in the docs.

**Comment by theNoahWilkes** [comment]
aha! I think that was in the debian instructions somewhere, I just skipped over it.  Working now.

**Reply by jef** [reply]
Awesome!
